src/common/ticketProvider/ticketProviderRoot.ts

```diff
@@ -25,9 +26,20 @@ export abstract class TicketProviderRoot {
   public async getDeploymentTag(): Promise<string> {
     const currentGitBranch = await getCurrentGitBranch() || "";
     let tag = currentGitBranch.toUpperCase() + "_DEPLOYED";
+
+    if (GitProvider.isDeployBeforeMerge()) {
+      const prInfo = await GitProvider.getPullRequestInfo();
+      const targetBranch = prInfo?.targetBranch || process.env.FORCE_TARGET_BRANCH;
+      if (targetBranch) {
+        tag = targetBranch.toUpperCase() + "_DEPLOYED";
+      }
+    }
+
     if (process.env?.DEPLOYED_TAG_TEMPLATE && !(process.env?.DEPLOYED_TAG_TEMPLATE || "").includes("$(")) {
-      tag = process.env?.DEPLOYED_TAG_TEMPLATE.replace("{BRANCH}", currentGitBranch.toUpperCase());
+      const branchToUse = tag.replace("_DEPLOYED", "");
```

Comment:
@0ptaq0  Why replacing "_DEPLOYED" ? 
If you calculated and sent your own tag, it should already  contain the string you want , no ? :)

Reply:
Before I added my part we already knew that the branch name is always `currentGitBranch.toUpperCase()`
But when I added the if statement it can be above mentioned or `targetBranch.toUpperCase()`

So in the end when I want to replace `{BRANCH}` I need to know with what I should replace it. That's the only reason why I am replacing _DEPLOYED, to get final branch name.

The other way I thought would be to override `currentGitBranch` with targetBranch in the added if statement. But I didn't want to change that part, as I am not sure of the whole sfdx-hardis structure.